src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,58 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  width: $calendar-width;
+  margin: auto;
+  padding: 70px 10px;
+
+  @for $index from 28 through 31 {
+    &--month-length_#{$index - 1}
+    &__day:nth-child(n + #{$index}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $day-width;
+    height: 100px;
+
+    background-color: #eee;
+    border: 1px solid black;
+
+    font-family: Arial, serif;
+    font-size: 30px;
+
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: aquamarine;
+      transform: translateY(-20px);
+    }
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+      }
+    }
+
+    $days:
+      mon 0,
+      tue ($day-width * 1 + $day-gap * 1),
+      wed ($day-width * 2 + $day-gap * 2),
+      thu ($day-width * 3 + $day-gap * 3),
+      fri ($day-width * 4 + $day-gap * 4),
+      sat ($day-width * 5 + $day-gap * 5),
+      sun ($day-width * 6 + $day-gap * 6);
```

Comment:
It is better to export these variables to a variable file

Reply:
Good point, fixed!